warehouse/scripts/dbt_artifacts.py

```diff
@@ -0,0 +1,280 @@
+"""
+Built off the starting point of https://guitton.co/posts/dbt-artifacts
+"""
+import json
+import os
+import yaml
+from datetime import datetime
+from enum import Enum
+from pathlib import Path
+from slugify import slugify
+from typing import Annotated, Any, ClassVar, Dict, List, Literal, Optional, Tuple, Union
+
+import pendulum
+from pydantic import BaseModel, Field
+from sqlalchemy import MetaData, Table, select
+
+
+class FileFormat(str, Enum):
+    csv = "csv"
+    geojson = "geojson"
+    json = "json"
+    jsonl = "jsonl"
+    mbtiles = "mbtiles"
+
+
+class DbtResourceType(str, Enum):
+    model = "model"
+    analysis = "analysis"
+    test = "test"
+    operation = "operation"
+    seed = "seed"
+    source = "source"
+
+
+class DbtMaterializationType(str, Enum):
+    table = "table"
+    view = "view"
+    incremental = "incremental"
+    ephemeral = "ephemeral"
+    seed = "seed"
+    test = "test"
+
+
+class NodeDeps(BaseModel):
+    macros: List[str]
+    nodes: List[str]
+
+    @property
+    def resolved_nodes(self) -> List["BaseNode"]:
+        return [BaseNode._instances[node] for node in self.nodes]
+
+
+class NodeConfig(BaseModel):
+    alias: Optional[str]
+    schema_: str = Field(None, alias="schema")
+    materialized: Optional[DbtMaterializationType]
+
+
+class Column(BaseModel):
+    name: str
+    description: Optional[str]
+    meta: Optional[Dict[str, Any]]
+    parent: "BaseNode" = None  # this is set after the fact
+
+    @property
+    def publish(self) -> bool:
+        return not self.meta.get("publish.ignore", False)
```

Comment:
lol at double negative, but I guess it makes sense

Reply:
yeah lol I think it makes more sense to explicitly flag columns to skip, and include them by default